### PR TITLE
Add zipswap adapter

### PR DESCRIPTION
**Pull request by rokinot**
Adding adapter for zipswap pools @ [zipswap.fi/](https://info.zipswap.fi/)

### CODE REVIEW

**Comment by 0xngmi** [comment]
hey, are you interested in applying for a position at defillama or integrating this adapter for other reasons?

**Comment by rokinot** [comment]
Interested in the position yes, but as of now I am underqualified, so take it as an independent contribution.

**Comment by slasher125** [comment]
hello rokinot. ty for adding! left a smol comment

File: src/adaptors/zipswap/index.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+`
+const getDate = async () => {
+  let data = await request(subgraphURL ,getCurrentDate);
+  let date = data.uniswapDayDatas[0].date;
+  return date+"";
+}
+
+var query = gql`
+{
+  pairDayDatas(first: 5,
```
Comment (slasher125) [inline src/adaptors/zipswap/index.js]: why only first 5? i'd increase that value cause u missing some pools, like 'WETH-ZIP'.

Comment (rokinot) [reply src/adaptors/zipswap/index.js]: Initially thought 5 would get the most relevant pools, but it the volume varies too much on different days. Expanded to 10 now to cover all relevant pools.

**Comment by llamatester** [comment]
The adapter at src/adaptors/zipswap/index.js exports pools: 
        
 
 
    
    Adaptor Runtime: 0.35 sec
    
    Running tests...
    
    Nb of pools: 10
     
    
    Sample pools:
    ┌─────────┬──────────────────────────────────────────────┬────────────┬───────────┬─────────────┬────────────────────┬────────────────────┐
    │ (index) │                     pool                     │   chain    │  project  │   symbol    │       tvlUsd       │        apy         │
    ├─────────┼──────────────────────────────────────────────┼────────────┼───────────┼─────────────┼────────────────────┼────────────────────┤
    │    0    │ '0x1a981daa7967c66c3356ad044979bc82e4a478b9' │ 'Optimism' │ 'zipswap' │ 'WETH-USDC' │ 1227643.7714345534 │ 11.680401632532062 │
    │    1    │ '0x53790b6c7023786659d11ed82ee03079f3bd6976' │ 'Optimism' │ 'zipswap' │ 'WETH-DAI'  │ 403995.9031403269  │ 7.295405112256733  │
    │    2    │ '0x3f6da9334142477718be2ecc3577d1a28dceaae1' │ 'Optimism' │ 'zipswap' │ 'gOHM-WETH' │ 256858.44913653258 │ 1.601332810170706  │
    │    3    │ '0x167dc49c498729223d1565df3207771b4ee19853' │ 'Optimism' │ 'zipswap' │  'WETH-OP'  │ 239485.75987367574 │ 11.05108762313259  │
    │    4    │ '0x251de0f0368c472bba2e1c8f5db5ac7582b5f847' │ 'Optimism' │ 'zipswap' │ 'WETH-WBTC' │  73006.6730505403  │ 3.227234578126813  │
    │    5    │ '0xd7f6ecf4371eddbd60c1080bfaec3d1d60d415d0' │ 'Optimism' │ 'zipswap' │ 'WETH-ZIP'  │ 35808.38089449239  │ 1.3432788221485041 │
    │    6    │ '0x73c31c30a9c9671d2b151c6765d80569b58fb976' │ 'Optimism' │ 'zipswap' │  'USDC-O3'  │ 2437.5198983738196 │ 3.6302378707625773 │
    │    7    │ '0x6b8ef8e744ad206d52f143560f49c1e5d60797b7' │ 'Optimism' │ 'zipswap' │ 'CRV-USDC'  │ 2215.5850125752254 │ 3.127558992433213  │
    │    8    │ '0xab81e7dbb9b0268d73cd61f3000f7c797f960695' │ 'Optimism' │ 'zipswap' │  'OP-USDC'  │ 227.0804845833024  │ 2.9390801319967417 │
    │    9    │ '0x9f0240fe9afde70c9d7e1d8b146d0b6e7cf84823' │ 'Optimism' │ 'zipswap' │ 'VELO-WETH' │  52.5175866364663  │ 14.753439190726164 │
    └─────────┴──────────────────────────────────────────────┴────────────┴───────────┴─────────────┴────────────────────┴────────────────────┘

**Pull request merged**